web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -177,19 +206,32 @@ const AppLaunch = ({ app }: { app: App }) => {
       </ButtonBorder>
     );
   }
+  function handleSamlAppEditButtonClick() {
+    setUpdateDialogOpen(true);
+    setResourceSpec({
+      name: name,
+      event: DiscoverEventResource.SamlApplication,
+      kind: ResourceKind.SamlApplication,
+      samlMeta: { preset: samlAppPreset },
+      icon: 'Application',
+      keywords: 'saml',
+    });
+  }
   if (samlApp) {
     return (
-      <ButtonBorder
-        as="a"
+      <ButtonWithMenu
+        text="Log In"
         width="123px"
         size="small"
         target="_blank"
         href={samlAppSsoUrl}
         rel="noreferrer"
         textTransform="none"
+        forwardedAs="a"
+        title="Log in to this SAML application"
```

Comment:
I'm not sure if `this` is needed.
```suggestion
        title="Log in to SAML application"
```